google/api_core/path_template.py

```diff
@@ -193,3 +226,62 @@ def validate(tmpl, path):
     """
     pattern = _generate_pattern_for_template(tmpl) + "$"
     return True if re.match(pattern, path) is not None else False
+
+def transcode(http_options, **request_kwargs):
+    """Transcodes a grpc request pattern into a proper HTTP request following the rules outlined here,
+       https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L44-L312
```

Comment:
Would it be helpful at all to use the compiled `http_pb2.py` representation in the common protos package? https://github.com/googleapis/python-api-common-protos/tree/main/google/api

Reply:
I think it's ok for us to what's in `http.proto` for now, and we can look to use the compiled representation in `http_pb2.py` in the future.